lib/src/Pos/Diffusion/Full/Block.hs

```diff
@@ -261,32 +267,34 @@ getBlocks logic enqueue nodeId tipHeader checkpoints = do
     -- required for the old version to correctly say OldestFirst).
     retrieveBlocksDo
         :: ConversationActions MsgGetBlocks MsgBlock d
+        -> BlockVersionData
         -> Int        -- ^ Index of block we're requesting
         -> [Block]    -- ^ Accumulator
         -> ExceptT Text d (NewestFirst [] Block)
-    retrieveBlocksDo conv !i !acc
+    retrieveBlocksDo conv bvd !i !acc
         | i <= 0    = pure $ NewestFirst acc
-        | otherwise = lift (recvLimited conv) >>= \case
+        | otherwise = lift (recvLimited conv (mlMsgBlock bvd)) >>= \case
               Nothing ->
                   throwError $ sformat ("Block retrieval cut short by peer at index #"%int) i
               Just (MsgNoBlock t) ->
                   throwError $ sformat ("Peer failed to produce block #"%int%": "%stext) i t
               Just (MsgBlock block) -> do
-                  retrieveBlocksDo conv (i - 1) (block : acc)
+                  retrieveBlocksDo conv bvd (i - 1) (block : acc)
 
 requestTip
     :: forall d t .
        ( DiffusionWorkMode d
-       , HasAdoptedBlockVersionData d
        )
-    => EnqueueMsg d
+    => Logic d
+    -> EnqueueMsg d
     -> (BlockHeader -> NodeId -> d t)
     -> d (Map NodeId (d t))
-requestTip enqueue k = enqueue (MsgRequestBlockHeaders Nothing) $ \nodeId _ -> pure . Conversation $
+requestTip logic enqueue k = enqueue (MsgRequestBlockHeaders Nothing) $ \nodeId _ -> pure . Conversation $
     \(conv :: ConversationActions MsgGetHeaders MsgHeaders m) -> do
         logDebug "Requesting tip..."
+        bvd <- getAdoptedBVData logic
         send conv (MsgGetHeaders [] Nothing)
-        received <- recvLimited conv
+        received <- recvLimited conv (mlMsgHeaders bvd 2200)
```

Comment:
2200?
Did you miss to use recoveryHeaderMessage?

Reply:
Oh yes, because we don't have the configuration value here. Which one did it come from? Infra, node, block? I think `HasBlockConfiguration`...  I suppose for the time being, this value is a configuration parameter for the full diffusion layer, so I'll put it back in.

Reply:
Oh, looks like I *do* already take it as a parameter. Just forgot to wire it up....